### PR TITLE
[FSDE-31007] Add search input + dropdown template

### PR DESCRIPTION
**Pull request by lumdmtr**
https://ctl.atlassian.net/browse/FSDE-31007

### CODE REVIEW

File: src/website/views/components/forms/search-input/_examples.pug [hunk 1]
```diff
@@ -116,6 +116,91 @@ p.-text
...
+          href=`#html-with-dropdown`
+          role="tab"
+          aria-selected="false"
+          tabindex="-1"
+          aria-controls="html-with-dropdown"
+        ) HTML Blueprint
+  .chi-tabs-panel.-active#webcomponent-with-dropdown
+    :code(lang="html")
+      <div class="chi-form__item">
+        <chi-search-input></chi-search-input>
```
Comment (jllr) [inline src/website/views/components/forms/search-input/_examples.pug:152]:
```suggestion
        <chi-search-input id="with-dropdown"></chi-search-input>
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 2]
```diff
@@ -390,5 +475,22 @@ script.
...
+      })
     }
   );
+
+
+style.
+  #dropdown .chi-dropdown__menu.-z--10.sc-chi-dropdown.sc-chi-dropdown-s.-active {
+    position: absolute;
+  }
+
```
Comment (jllr) [inline src/website/views/components/forms/search-input/_examples.pug:496]:
```suggestion


```

Comment (jllr) [reply src/website/views/components/forms/search-input/_examples.pug:496]: @lumdmtr I have removed in codepen file and it worked fine. I think is not neccessary

**Review by jllr — CHANGES_REQUESTED** [review]

File: src/website/views/components/forms/search-input/_examples.pug [hunk 3]
```diff
@@ -116,6 +116,91 @@ p.-text
...
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 3</a>
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 4</a>
+        </chi-dropdown>
+      </div>
+
+      <script>
+        const input = document.getElementById("with-dropdown");
+
+        input.addEventListener("chiInput", () => {
+          document.getElementById("dropdown").show(); // toggle()
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:165]:
```suggestion
          dropdown.show();
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 4]
```diff
@@ -116,6 +116,91 @@ p.-text
...
+        <chi-dropdown id="dropdown" prevent-auto-hide="" position="bottom">
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 1</a>
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 2</a>
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 3</a>
+          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 4</a>
+        </chi-dropdown>
+      </div>
+
+      <script>
+        const input = document.getElementById("with-dropdown");
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:162]: Add the `chiClean` event to hide the dropdown.

```
input.addEventListener("chiClean", () => {
  dropdown.hide();
});
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 5]
```diff
@@ -390,5 +475,22 @@ script.
     'DOMContentLoaded',
     function() {
       chi.tab(document.querySelectorAll('.chi-tabs-panel .chi-tabs'));
+
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:478]: Make the changes requested for the web components code snippet here as well.

File: src/website/views/components/forms/search-input/_examples.pug [hunk 6]
```diff
@@ -116,6 +116,91 @@ p.-text
...
+        })
+      </script>
+  .chi-tabs-panel#html-with-dropdown
+    .chi-tab__description
+      span.-text--grey
+        | This HTML Blueprint requires JavaScript.
+        | You may use your own solution, or use Chi's vanilla JavaScript solution,
+        | <a href="../../getting-started/installation">Chi.js</a>.
+    :code(lang="html")
+      <div class="chi">
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:179]: Review the HTML code snippet and make sure that it works.

File: src/website/views/components/forms/search-input/_examples.pug [hunk 7]
```diff
@@ -116,6 +116,91 @@ p.-text
...
+        .chi-form__item
+          chi-search-input(id="with-dropdown")
+          chi-dropdown#dropdown(prevent-auto-hide='' position='bottom')
+            a.chi-dropdown__menu-item(href='#' slot='menu') Item 1
+            a.chi-dropdown__menu-item(href='#' slot='menu') Item 2
+            a.chi-dropdown__menu-item(href='#' slot='menu') Item 3
+            a.chi-dropdown__menu-item(href='#' slot='menu') Item 4
+
+  .example-tabs.-pl--2
+    ul.chi-tabs#example-with-dropdown(role="tablist")
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:133]: Include a Vue example

File: src/website/views/components/forms/search-input/_examples.pug [hunk 8]
```diff
@@ -116,6 +116,88 @@ p.-text
...
+        a(
+          href=`#html-with-dropdown`
+          role="tab"
+          aria-selected="false"
+          tabindex="-1"
+          aria-controls="html-with-dropdown"
+        ) HTML Blueprint
+  .chi-tabs-panel.-active#webcomponent-with-dropdown
+    :code(lang="html")
+      <div class="chi chi-form__item">
```
Comment (jllr) [inline src/website/views/components/forms/search-input/_examples.pug:151]:
```suggestion
      <div class="chi-form__item">
```

**Review by jllr — CHANGES_REQUESTED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

File: src/website/views/components/forms/search-input/_examples.pug [hunk 9]
```diff
@@ -116,6 +116,96 @@ p.-text
...
       </div>
 
+h3 With Dropdown
+.example.-mb--3
+  .-p--3
+    .chi-grid
+      .chi-col.-w--12.-w-sm--8.-w-md--6.-w-lg--5
+        .chi-form__item
+          chi-search-input#example__with_dropdown
+          chi-dropdown#example__dropdown(position='bottom')
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:126]: Include here and in the code snippets.

```suggestion
          chi-dropdown#example__dropdown(prevent-auto-hide position='bottom')
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 10]
```diff
@@ -390,5 +480,24 @@ script.
...
+        dropdown.show();
+      });
+
+      input.addEventListener("chiBlur", () => {
+        dropdown.hide();
+      });
+
+      input.addEventListener("chiClean", () => {
+        dropdown.hide();
+      });
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:501]:
```suggestion
      input.addEventListener("chiFocus", () => {
        dropdown.show();
      });

      input.addEventListener("chiBlur", () => {
        dropdown.hide();
      });
```

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**